### PR TITLE
Move CLI into its own module (#316)

### PR DESCRIPTION
**Pull request by whimboo**
This PR fixes issue #316. @davehunt mind reviewing it?

### CODE REVIEW

File: mozdownload/cli.py [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+import sys
+
+from . import errors
+from . import scraper
+
+
+version = pkg_resources.require("mozdownload")[0].version
+
+__doc__ = """
+Module to handle downloads for different types of archive.mozilla.org hosted \
```
Comment (AutomatedTester) [inline mozdownload/cli.py]: the `\` isnt needed

File: mozdownload/cli.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    parser.add_option('--url',
+                      dest='url',
+                      metavar='URL',
+                      help='URL to download. Note: Reserved characters (such '
+                           'as &) must be escaped or put in quotes otherwise '
+                           'CLI output may be abnormal.')
+    parser.add_option('--version', '-v',
+                      dest='version',
+                      metavar='VERSION',
+                      help='Version of the application to be used by release\
```
Comment (AutomatedTester) [inline mozdownload/cli.py]: You have used `\` and silent concat (e.g. `"foo" "bar"` == "foobar"`). Please try be consistent.

File: mozdownload/cli.py (hunk 1 above)
Comment (whimboo) [inline mozdownload/cli.py]: That's an old relict and hasn't been newly added.  Would you be ok when I fix that with my next PR? It's already fixed there locally.

File: mozdownload/cli.py (hunk 2 above)
Comment (whimboo) [inline mozdownload/cli.py]: That's something I would also do in the argparse PR given that I would have to touch those lines anyway.

File: mozdownload/cli.py (hunk 1 above)
Comment (whimboo) [inline mozdownload/cli.py]: Just to add... reason why those instances are present is that I only copy and pasted the code from one file to the other to minimize code changes for the reviewer. Sorry, that I didn't explicitly mention that.

**Comment by AutomatedTester** [comment]
If you are doing it in a follow up then fine. r+

**Pull request merged**